### PR TITLE
Blocking and Staging logic

### PR DESCRIPTION
**Pull request by matentzn**
This pull implements the blocking and staging logics, which are totally independent.
- Blocking is implemented through cypher queries, see process.sh lines 56-57.
- Staging is the rest. We only care here about the embargo logic of staging. So if we embargo, then, depending on whether we are in the prod or dev stage mode (see Dockerfile), we will embargo different things. The logic corresponds to what was discussed [here](https://github.com/VirtualFlyBrain/neo4j2owl/issues/52). The implementation is realised through two different sets of sparql queries (one for prod, one for dev), which apply differently rigorous embargo rules. The prod queries are unchanged, and the embargo rules in dev are tighter (i.e. less stuff gets embargoed).

see #8
see https://github.com/VirtualFlyBrain/neo4j2owl/issues/52

### CODE REVIEW

**Review by Robbie1977 — APPROVED** [review]

File: sparql/select_blocked_entities.sparql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
```
Comment (dosumis) [inline sparql/select_blocked_entities.sparql:1]: Presumably this SPARQL no longer needed/used?

File: sparql/select_blocked_entities.sparql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
```
Comment (dosumis) [inline sparql/select_blocked_entities.sparql:2]: SPARQL query no longer needed?

File: sparql/select_blocked_relations.sparql [hunk 3]
```diff
@@ -0,0 +1,22 @@
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
```
Comment (dosumis) [inline sparql/select_blocked_relations.sparql:1]: Sparql query longer needed?

File: process.sh [hunk 4]
```diff
@@ -53,24 +53,28 @@ date
 echo '** Exporting KB to OWL **'
 curl -i -X POST ${KBserver}/db/data/transaction/commit -u ${KBuser}:${KBpassword} -H 'Content-Type: application/json' -d '{"statements": [{"statement": "MATCH (c) REMOVE c.label_rdfs RETURN c"}]}' >> ${VFB_DEBUG_DIR}/neo4j_remove_rdfs_label.txt
 curl -i -X POST ${KBserver}/db/data/transaction/commit -u ${KBuser}:${KBpassword} -H 'Content-Type: application/json' -d '{"statements": [{"statement": "MATCH (p) WHERE EXISTS(p.label) SET p.label_rdfs=[] + p.label"}]}' >> ${VFB_DEBUG_DIR}/neo4j_change_label_to_rdfs.txt
+curl -i -X POST ${KBserver}/db/data/transaction/commit -u ${KBuser}:${KBpassword} -H 'Content-Type: application/json' -d '{"statements": [{"statement": "MATCH (n:Entity) WHERE exists(n.block) DETACH DELETE n"}]}' >> ${VFB_DEBUG_DIR}/neo4j_change_label_to_rdfs.txt
+curl -i -X POST ${KBserver}/db/data/transaction/commit -u ${KBuser}:${KBpassword} -H 'Content-Type: application/json' -d '{"statements": [{"statement": "MATCH ()-[r]-() WHERE exists(r.block) DELETE r"}]}' >> ${VFB_DEBUG_DIR}/neo4j_change_label_to_rdfs.txt
```
Comment (dosumis) [inline process.sh:57]: We might want to consider this (although happy to make this a separate ticket for Robbie & myself to consider):

```cql
MATCH (n:Entity) WHERE apoc.convert.toBoolean(ds.block[0]) DETACH DELETE n
MATCH ()-[r]->() WHERE apoc.convert.toBoolean(ds.block[0]) DELETE r
```
This allows for block to explicitly be set to [False]
The absence of a key => False
['Some string'] => True  (unless this is the string 'false') # This allows comments about obsoletion to be stored here and case as True - something Robbie has requested.

Weakness:  pipeline is brittle to value not being a list, although this is currently enforced by kb2kb.

**Review by dosumis — APPROVED** [review]
Happy to merge - but have commented a bit.

File: process.sh (hunk 4 above)
Comment (Robbie1977) [reply process.sh:57]: I'm happy we shouldn't have a block parameter unless we actually want to block as no reason to encourage block=false being there for no reason.

**Pull request merged**